### ISSUE TITLE
Backwards-incompatible nn.GroupNorm in torch 1.10

### ISSUE THREAD

**Issue by anton-l** [pytorch/pytorch#67907]
## 🐛 Bug

When comparing the outputs of `nn.GroupNorm()` when used with a uniform (i.e. `std = 0`) input, the outputs differ between `torch==1.9` and `torch==1.10`, and that difference is amplified by the downstream network layers, breaking reproducibility.

This probably happens due to recent changes to the normalization algorithm: https://github.com/pytorch/pytorch/issues/40302 & https://github.com/pytorch/pytorch/pull/54921

## To Reproduce

`$ pip install torch==1.9.1`
```python
import torch

inputs = torch.ones((1,32,25), dtype=torch.float32) * 0.05
group_norm = torch.nn.GroupNorm(32, 32, affine=True)

print(group_norm(inputs).sum())

>>> tensor(0.0017, grad_fn=<SumBackward0>)
```
`$ pip install torch==1.10`

```python
import torch

inputs = torch.ones((1,32,25), dtype=torch.float32) * 0.05
group_norm = torch.nn.GroupNorm(32, 32, affine=True)

print(group_norm(inputs).sum())

>>> tensor(0.0003, grad_fn=<SumBackward0>)
```

## Expected behavior

Having `tensor(0.0017, grad_fn=<SumBackward0>)` as the result of running the same script after upgrading to `torch==1.10`

## Environment
Tested in a Google Colab environment.
```
PyTorch version: 1.10.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.12.0
Libc version: glibc-2.26

Python version: 3.7.12 (default, Sep 10 2021, 00:21:48)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: True
CUDA runtime version: 11.1.105
GPU models and configuration: GPU 0: Tesla K80
Nvidia driver version: 460.32.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.0.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] torch==1.10.0
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.10.0
[pip3] torchvision==0.10.0+cu111
```
## Additional context

The incompatibility was discovered due to different outputs in `huggingface/transformers` `Wav2Vec2Model` when testing it with a uniform input (e.g. `np.ones()`): https://github.com/huggingface/transformers/pull/14169#issuecomment-954218308

cc @albanD @mruberry @jbschlosser @walterddr

**Comment by gchanan** [comment]
The reproducibility documentation says PyTorch don't support numerical reproducibility in this case (https://pytorch.org/docs/stable/notes/randomness.html):
> Completely reproducible results are not guaranteed across PyTorch releases, individual commits, or different platforms.

**Comment by patrickvonplaten** [comment]
Also cc'ing @alexeib here for information - this seems to have quite a significant effect on all wav2vec2 and hubert base models...
